product_multi_barcode_constraint_per_company/tests/test_product_barcode_constraint.py

```diff
@@ -0,0 +1,148 @@
+        self.product_test_1.barcode = self.valid_barcode_1
+        self.assertEqual(len(self.product_test_1.barcode_ids), 1)
+        self.assertEqual(
+            self.product_test_1.barcode_ids.name, self.product_test_1.barcode
+        )
+        self.product_test_1.company_id = self.company_1.id
+
+        self.product_test_2.company_id = self.company_1.id
+        with mute_logger("odoo.sql_db"), self.assertRaises(
+            UserError,
+            msg=_(
+                'The Barcode "%(barcode_name)s" already exists for '
+                'product "%(product_name)s" in the company %(company_name)s'
+            )
+            % dict(
+                barcode_name=self.valid_barcode_1,
+                product_name=self.product_test_1.name,
+                company_name=self.product_test_1.company_id.name,
+            ),
+        ):
+            self.product_test_2.barcode = self.valid_barcode_1
+
+    def test_barcodes(self):
+        self.product_test_1.barcode = self.valid_barcode_1
+        self.product_test_1.company_id = self.company_1.id
+        message = (
+            _(
+                'The Barcode "%(barcode_name)s" already exists for '
+                'product "%(product_name)s" in the company %(company_name)s'
+            )
+            % dict(
+                barcode_name=self.valid_barcode2_1,
+                product_name=self.product_test_2.name,
+                company_name=self.product_test_2.company_id.name,
+            ),
+        )
+        with mute_logger("odoo.sql_db"), self.assertRaises(UserError, msg=message):
+            self.product_test_2.barcode_ids = [
+                (0, 0, {"name": self.valid_barcode2_1}),
+                (0, 0, {"name": self.valid_barcode2_1}),
+            ]
+
+        self.product_test_2.company_id = self.company_1.id
+        with mute_logger("odoo.sql_db"), self.assertRaises(UserError, msg=message):
+            self.product_test_2.barcode_ids = [
+                (0, 0, {"name": self.valid_barcode2_1}),
+                (0, 0, {"name": self.valid_barcode2_1}),
+            ]
+        self.product_test_2.company_id = self.company_2.id
+        with mute_logger("odoo.sql_db"), self.assertRaises(UserError, msg=message):
+            self.product_test_2.barcode_ids = [
+                (0, 0, {"name": self.valid_barcode2_1}),
+                (0, 0, {"name": self.valid_barcode2_1}),
+            ]
+        with mute_logger("odoo.sql_db"), self.assertRaises(UserError, msg=message):
+            self.product_test_2.barcode_ids = [
+                (0, 0, {"name": self.valid_barcode2_1}),
+                (0, 0, {"name": self.valid_barcode_1}),
+            ]
+
+        with mute_logger("odoo.sql_db"), self.assertRaises(UserError, msg=message):
+            self.product_test_2.barcode_ids = [
+                (0, 0, {"name": self.valid_barcode_1}),
+                (0, 0, {"name": self.valid_barcode2_1}),
+            ]
```

Comment:
(non-blocking)

In case you didn't know, you can decorate the entire test method with `mute_logger("odoo.sql_db")` so you don't have to specify it every time here.

Reply:
Thanks, I'll fix it